### ISSUE TITLE
Is the project still being maintained?

### ISSUE THREAD

**Issue by panoukos41** [lbugnion/mvvmlight#69]
I know i can get the project and make my own version while fixing issues but some issues have been resolved by other members and they made pull requests and i am curious if the project is still alive or if i should start making changes and use it outside nuget(which i do not prefer :P).

**Comment by lbugnion** [comment]
Hi,

This is a fair question and to be honest I have been so overloaded with Azure work since I joined Microsoft that it has been difficult for me to keep up with MVVM Light. I am in the process of finding solutions to this issue.

Sorry for the inactivity.

Take care
Laurent

**Comment by MichaelShapiro** [comment]
Thank you for a great question and I appreciate Laurent's quick response. I am in the same boat - deciding if I should take it outside of the nuget or rely on your continuing development. 

Laurent - It was your vision and implementation and I think there is no better person than you who can keep and evolve this project. The biggest road block for me is that the new NETstandard version does not work with Droid projects that rely on AndroidSupport library. Yes, I already posted this question and even emailed you and Jim:) It would be extremely helpful if you could comment on it or offer some directions on how to alleviate this situation. I really rally hope that I can avoid changing the core of your code.

Thank you very much,
mike

**Comment by panoukos41** [comment]
Thanks a lot for the fast response its good to know that you are still here and looking at issues, don't worry i believe everyone knows that you dont have much free time to really work here and you need some free time to relax too ^_^.
I was just curious since i started a small library that extends some things from mvvmlight and i really want to contribute them in the future when I learn more about nuget etc.
I look forward hearing from you and your solutions in the near future for the project when you have the time :D !

Thank you very much,
Panos

**Comment by NPadrutt** [comment]
Is there any update on this?

**Comment by Keboo** [comment]
@lbugnion I have been a big fan of this library for several years and would love to see it continue. If you are interested in finding people to help maintain it I would be willing to help out.

**Comment by jamesmcroft** [comment]
@Keboo @lbugnion I think there a few of us who would enjoy maintaining this project. Because of Android support libraries in Xamarin, I've already forked this repo to produce my own maintained NuGet package.

It would be great to see this handed over or alternatives being recommended for migrating away.

**Comment by mscottbarnes** [comment]
It's been in the process of finding support for 8 months now.  And it's not like the need is going away.  Can you also update the samples?  I wonder if jamesmcroft's package has any samples.  He doesn't seem so tied up in Azure.

**Comment by jamesmcroft** [comment]
@mscottbarnes my fork of the repo doesn't have any samples at the moment, only the updated source code. It's not for the sake of not wanting to produce any, it's that I'd prefer not to have to maintain my own fork and NuGet packages of this incredible MVVM toolkit. 

To be honest, I'm more than open to maintaining and building a community around MvvmLight and opening it up more contributions. Would need @lbugnion to be open to this though and bringing more folks in.

**Comment by NPadrutt** [comment]
since the last update of @lbugnion  in this repository was over a year ago, I kinda wonder if this project just got forgotten and if there actually will be a transition to a new maintainer..

**Comment by jamesmcroft** [comment]
@NPadrutt I'm keen to see what comes of the Windows Community Toolkit's basic MVVM toolkit which is inspired by this project (https://github.com/windows-toolkit/WindowsCommunityToolkit/issues/3230)

However, I don't think it's going to be a full port of all components so there is a concern around the Xamarin Android and iOS support as MvvmLight provides a layer for bindings. It would be great to hear from Laurent and see if he's open to letting a new community pick this up and run with it. 

This project has been used as a dependency for many projects for the simplicity of building with it. That and it's very lightweight in comparison to other MVVM frameworks. It would definitely be sad to see it no longer maintained.

**Comment by michael-hawker** [comment]
@NPadrutt @jamesmcroft yup, we've been talking to @lbugnion about the new MVVM package in the Toolkit. We're moving forward with the toolkit implementation and should have a preview NuGet package out and some samples for it in July.

Ideally we'll have some samples for UWP, WPF, Xamarin and other platforms as well. I'm hoping there won't be as much need for any framework-specific packages/glue. We'll also try and provide some migration guides for folks coming from MVVM Light. @jamesmcroft I'd love some of your input/guidance/help with your expertise on MVVM Light about the migration guide, let's chat on Discord. Thanks!

**Comment by MichaelShapiro** [comment]
@michael-hawker Thank you for the update. How to find you on Discord?

I've never used the Toolkit, so I really do not know much about it. I hope I am not asking a stupid question, my apologies if I am.  Currently I use MVVMLight for Xamarin iOS and Xamarin Android apps. I am considering transitioning and migrating the development to Xamarin Forms. From what I briefly researched (once again, sorry if my question is stupid) Xamarin Forms do provide mvvm pattern right out of the box. So, is there really a need for external mvvm tool - MVVM Light, or what you are building  within the Toolkit -  for Xamarin Forms based project?

Thank you,
mike

**Comment by Sergio0694** [comment]
Hey @MichaelShapiro - sorry for the late reply!

I've looked at the Xamarin Forms docs on MVVM ([here](https://docs.microsoft.com/en-us/xamarin/xamarin-forms/enterprise-application-patterns/mvvm)), and it does seem like they're providing some basic implementations for types needed by the MVVM pattern out of the box (eg. [`BindableObject`](https://docs.microsoft.com/en-us/dotnet/api/xamarin.forms.bindableobject?view=xamarin-forms), [`Command`](https://docs.microsoft.com/en-us/dotnet/api/xamarin.forms.command?view=xamarin-forms), etc.). So to answer your question - yes, in theory you could write a completely self-contained Xamarin.Forms app using the MVVM pattern by just using the APIs provided there.

I wouldn't generally recommend doing that though compared to writing your MVVM backend using the MVVM Toolkit (full disclosure, I might be a bit biased here as I'm the author of that, but I do think there is a valid argument to be made here). The issue is that by relying on those Xamarin.Forms APIs, you're creating a strong dependency on those assemblies in your entire MVVM backend, which doesn't help to make your code modular. You'd effectively be forced to either just put all your MVVM backend right into your Xamarin.Forms project, or into some external project that is still targeting Xamarin.Forms. I would argue this is not the best approach both to deal with things such as unit testing (you'd need a framework-specific unit test project instead of being able to just use a general .NET Core unit test project), as well as to create a future-proof codebase in case you ever wanted to write a frontend of your app for anything from a native framework (eg. UWP), webassembly, or literally anything else. I do believe that the best solution for MVVM backends is for them to just target .NET Standard (or .NET 5 in the future).

We'll be publishing a new preview version of the MVVM Toolkit package in the near future, and we're also working on updating all the docs and samples we have over at https://aka.ms/mvvmtoolkit. If you decide to try out the new lib, let us know how it goes! I have a tracking issue to receive early feedbacks [here](https://github.com/windows-toolkit/WindowsCommunityToolkit/issues/3428), while the library is still in preview. Hope this helps! 😄

**Comment by NPadrutt** [comment]
@Sergio0694 is there any ETA for the new MVVM Toolkit?

**Comment by Sergio0694** [comment]
@NPadrutt The official release of the MVVM Toolkit will be part of the 7.0 release of the Windows Community Toolkit, which should be available sometime around November. We also already have a preview package up on NuGet ([here](https://www.nuget.org/packages/Microsoft.Toolkit.Mvvm/)), with a new version (which will contain some API changes, so you might want to wait for that one), probably coming within the next couple weeks.

**Comment by michael-hawker** [comment]
@lbugnion now that we've shipped the [MVVM Toolkit](https://aka.ms/mvvmtoolkit), what were your plans for this repo?

Migration Guide for those that are interested: https://docs.microsoft.com/windows/communitytoolkit/mvvm/migratingfrommvvmlight
